renku/api/storage.py

```diff
@@ -88,24 +88,30 @@ def track_paths_in_storage(self, *paths):
                 stderr=STDOUT,
                 cwd=str(self.path),
             )
-        elif self.use_external_storage:
+        else:
             raise errors.ExternalStorageNotInstalled(self.repo)
 
     def untrack_paths_from_storage(self, *paths):
         """Untrack paths from the external storage."""
-        if self.use_external_storage and self.external_storage_installed:
+        if not self._use_lfs():
+            return
```

Comment:
Usage of this function (places like remove and move) is computing certain path indexes which are not needed in this case. We should somehow detect that it's not needed to compute those things and not invoke this function at all if there is no storage from which we need to track/untrack.

Then we don't need this check at all.

Reply:
I rather keep the scope of this fix limited and would postpone that refactoring to a later time. Moreover, since this is a public function it's good to have this test in place just a safety-net.